File: leetcode/top-interview-questions-easy/strings/5. valid_palindrom.js
var isValid = function (char) {
  return /[0-9a-zA-Z]/.test(char);
};
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function (s) {
  var start = 0;
  var end = s.length - 1;
  while (start < end) {
    var started = s[start];
    var ended = s[end];
    while (start < end && !isValid(started)) {
      start++;
      started = s[start];
    }
    while (start < end && !isValid(ended)) {
      end--;
      ended = s[end];
    }
    if (started.toLowerCase() !== ended.toLowerCase()) {
      return false;
    }
    start++;
    end--;
  }

  return true;
};

File: leetcode/top-interview-questions-easy/strings/5. valid_palindrom.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        start = 0
        end = len(s) - 1
        while start < end:
            started = s[start]
            ended = s[end]
            while start < end and not started.isalnum():
                start+=1
                started=s[start]
            
            while start < end and not ended.isalnum():
                end-=1
                ended = s[end]
            
            if started.lower() != ended.lower():
                return False
            
            start+=1
            end-=1
        return True